lib/progress.ts

```diff
@@ -160,3 +160,21 @@ export const getLessonKey = (chapterId, lessonId) => {
 
   return lesson.metadata.key
 }
+
+export const getChapterFragment = (
+  userProgressKey: string,
+  forceFragment?: boolean
+): string => {
+  const progressHrefFragement =
+    userProgressKey === keys[keys.length - 1]
+      ? (parseInt(userProgressKey.substring(2, 3)) + 1).toString()
+      : userProgressKey.substring(2, 3)
+  if (
+    !userProgressKey ||
+    (userProgressKey.startsWith('CH1') && !forceFragment)
+  ) {
+    return ''
+  } else {
+    return `#chapter-${progressHrefFragement}`
+  }
```

Comment:
We can simplify this logic.

1. In the function definition, we explicitly define `userProgressKey` to be a string. So to again check for it being defined is redundant.
2. The name 'forceFragment' can be changed to 'getFragment' to sound closer to its usage.

```suggestion
if (getFragment || !userProgressKey.startsWith('CH1')) {
  return `#chapter-${progressHrefFragement}`
}
return ''
```

Reply:
I simplified the logic such that I do not think a comment is necessary to describe is function.